src/Sylius/Bundle/ApiBundle/Serializer/Normalizer/GeneratedPromotionCouponsNormalizer.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Serializer\Normalizer;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Sylius\Bundle\ApiBundle\SectionResolver\AdminApiSection;
+use Sylius\Bundle\CoreBundle\SectionResolver\SectionProviderInterface;
+use Sylius\Component\Core\Model\PromotionCouponInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareTrait;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+use Webmozart\Assert\Assert;
+
+final class GeneratedPromotionCouponsNormalizer implements NormalizerInterface, NormalizerAwareInterface
+{
+    use NormalizerAwareTrait;
+
+    private const ALREADY_CALLED = 'sylius_generated_promotion_coupons_normalizer_already_called';
+
+    public function __construct(private SectionProviderInterface $sectionProvider)
+    {
+    }
+
+    public function normalize($object, $format = null, array $context = [])
+    {
+        Assert::isInstanceOf($object, ArrayCollection::class);
+        Assert::isInstanceOf($object->first(), PromotionCouponInterface::class);
+        Assert::keyNotExists($context, self::ALREADY_CALLED);
+        Assert::isInstanceOf($this->sectionProvider->getSection(), AdminApiSection::class);
```

Comment:
```suggestion
```

Reply:
Why?

Reply:
This was already checked is `supportsNormalization` method. And `sectionProvider` it is not used in normalize method